odl/solvers/scalar/steplen.py

```diff
@@ -148,6 +148,21 @@ def __call__(self, x, direction, dir_derivative):
             The computed step length
         """
         alpha = 1.0
+
+        # Since all comparisons with nan returns False, in order to not get a
+        # problem with nan-values we throw an error.
```

Comment:
Good to have a comment. A bit twisted sentence though. Perhaps (slightly shorter)
```python
# Cannot meaningfully compare to NaN -> raise an exception
```